regression/pages/common/utils.py

```diff
@@ -0,0 +1,496 @@
+        return self.get_first_text_block(email_message)
+
+    @staticmethod
+    def get_first_text_block(email_message_instance):
+        """
+        In case of multiple payloads return first text block
+        Args:
+            email_message_instance:
+        """
+        maintype = email_message_instance.get_content_maintype()
+        if maintype == 'multipart':
+            for part in email_message_instance.get_payload():
+                if part.get_content_maintype() == 'text':
+                    return part.get_payload()
+        elif maintype == 'text':
+            return email_message_instance.get_payload()
+
+
+def get_target_url_from_text(partial_url_string, text_chunk):
+    """
+    Search and return the target url from text chunk, the url is searched
+    on the basis of partial string embedded in url
+    Args:
+        partial_url_string:
+        text_chunk:
+    Returns:
+        target url:
+    """
+    pattern = r"(?P<url>http[s]?://[^\s]+(/{}/)[^\s]+)".format(
+        partial_url_string
+    )
+    regex_result = re.search(pattern, text_chunk)
+    if regex_result:
+        target_url = regex_result.group("url")
+        return target_url.rstrip('.')
+    else:
+        return 'Target URL not found in the text'
+
+
+class EcommerceApiClient(object):
+    """
+    Ecommerce API client for various actions like creating, deleting
+    downloading coupons
+    """
+
+    @property
+    def ecommerce_api_client(self):
+        """
+        E-Commerce API Client
+        """
+        return EdxRestApiClient(
+            ECOMMERCE_API_URL, oauth_access_token=ACCESS_TOKEN)
+
+    def create_coupon(self, payload):
+        """
+        Create coupons using API
+        Args:
+            payload:
+        """
+        response = self.ecommerce_api_client.coupons.post(data=payload)
+        return response['coupon_id']
+
+    def get_coupon_report(self, coupon_id):
+        """
+        Get coupon report using api
+        Args:
+            coupon_id:
+        """
+        return self.ecommerce_api_client.products(coupon_id).get()
+
+    def delete_coupon(self, coupon_id):
+        """
+        Delete the coupon created in the tests and all associated stock
+        records and vouchers.
+        Args:
+            coupon_id:
+        """
+        self.ecommerce_api_client.coupons(coupon_id).delete()
+
+    def get_coupon_codes(self, coupon_id):
+        """
+        Get coupon codes from coupon report
+        Args:
+            coupon_id:
+        """
+        coupon_codes = []
+        items = (
+            self.get_coupon_report(coupon_id)['attribute_values'][0]['value'])
+        for item in items:
+            coupon_codes.append(item['code'])
+        return coupon_codes
+
+
+def yesterday_date():
+    """
+    Get and return yesterday's date in dd-mmm-yyyy format
+    """
+    return (datetime.date.today() - datetime.timedelta(1)).strftime("%d-%b-%Y")
```

Comment:
the emails are generated during the tests so there is a lot of buffer for time zone differences